code/drasil-database/lib/Database/Drasil/ChunkDB.hs

```diff
@@ -191,15 +191,15 @@ makeLenses ''ChunkDB
 --     * 'TheoryModel's (for 'TheoryModelMap'),
 --     * 'ConceptInstance's (for 'ConceptInstanceMap'),
 --     * 'LabelledContent's (for 'LabelledContentMap').
-cdb :: (Quantity q, MayHaveUnit q, Idea t, Concept c, IsUnit u) =>
-    [q] -> [t] -> [c] -> [u] -> [DataDefinition] -> [InstanceModel] ->
+cdb :: (Quantity q, MayHaveUnit q, Concept c, IsUnit u) =>
+    [q] -> [IdeaDict] -> [c] -> [u] -> [DataDefinition] -> [InstanceModel] ->
```

Comment:
4. Constraining to the `Idea t => t` was kind of unnecessary because it was exclusively filled out with `IdeaDict`s, so here we are just removing it so that we know which monomorphic type to cache the `DataDefinition`'s terms as.

Reply:
If I was sure that we were going to keep `cdb` as is, I would predict that you'd come to regret this change. I was actually going to suggest that we move some other routines in the exact opposite direction!

Philosophically what you're doing is moving to a specific representation (`IdeaDict`) where before this worked based on an interface (`Idea t`). From an SE perspective, working to interfaces is better (except for efficiency, but not even here) than to specific implementations.